Userland/Libraries/LibVT/TerminalWidget.cpp

```diff
@@ -752,13 +752,12 @@ void TerminalWidget::paste()
     if (m_ptm_fd == -1)
         return;
 
-    auto mime_type = GUI::Clipboard::the().mime_type();
-    if (!mime_type.starts_with("text/"))
+    auto data_and_type = GUI::Clipboard::the().data_and_type();
+    if (!data_and_type.mime_type.starts_with("text/"))
         return;
-    auto text = GUI::Clipboard::the().data();
-    if (text.is_empty())
+    if (data_and_type.data.is_empty())
         return;
-    send_non_user_input(text);
+    send_non_user_input(data_and_type.data);
```

Comment:
could use structured bindings here to make this a bit nicer:

```cpp
auto [data, mime_type, _] = GUI::Clipboard::the().data_and_type();
if (!mime_type.starts_with("text/"))
    return;
if (data.is_empty())
    ...
```

probably applies elsewhere as well, I'm not going to point them all out individually.

Reply:
Wouldn't that copy everything? `auto&& [data, mime_type, _] = GUI::Clipboard::the().data_and_type();` is also permitted, I think that would move it instead.

Reply:
uh, guess you'd do `auto& [...]` then :)

Reply:
That wouldn't work, and I'm also wrong about needing `&&`. :D

Reply:
why not?

https://github.com/SerenityOS/serenity/blob/66c06e8efb7c5e2abbf81fdfccaa80f24a4edbe0/Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp#L362

Reply:
`GlyphEditorWidget.cpp:87:11: error: non-const lvalue reference to type 'GUI::Clipboard::DataAndType' cannot bind to a temporary of type 'GUI::Clipboard::DataAndType'`

As far as I understand, the returned DataAndType is a temporary, whereas `Optional::value()` returns an lvalue (`FunctionAndArgumentIndex&`).

Reply:
I see. Oh well, can't have all the nice things :^)